File: 2303. Calculate Amount Paid in Taxes/solution.cpp
class Solution {
public:
  double calculateTax(vector<vector<int>>& brackets, int income) {
    double result = 0;
    double previous = 0;
    for (int index = 0; index < brackets.size(); index++){
      vector<int> bracket = brackets[index];
      double upper = bracket[0];
      double percent = bracket[1];
      if (income >= upper){
        result += (upper - previous) * percent / 100;
        previous = upper;
      }
      else{
        result += (income - previous) * percent / 100;
        break;
      }
    }
    return result;
  }
};

File: 2303. Calculate Amount Paid in Taxes/solution.js
/**
 * @param {number[][]} brackets
 * @param {number} income
 * @return {number}
 */
 var calculateTax = function(brackets, income) {
  let result = 0;
  let previous = 0;
  for (let index = 0; index < brackets.length; index++){
    let bracket = brackets[index];
    let upper = bracket[0];
    let percent = bracket[1];
    if (income >= upper){
      result += (upper - previous) * percent / 100;
      previous = upper;
    }
    else{
      result += (income - previous) * percent / 100;
      break;
    }
  }
  return result;
};